src/simulator/simulation_parameters.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2021 The Private Cardinality Estimation Framework Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Defines the execution environment of a simulation instance."""
+
+from numpy.random import Generator
+from typing import List
+from typing import NamedTuple
+
+
+class LiquidLegionsParameters(NamedTuple):
+    """Simulation parameters specific to Liquid Legions.
+
+    decay_rate:  Decay rate parameter of Liquid Legions
+        sketch (often abbreviated as a).
+    sketch_size:  Size of Liquid Legions sketch (often
+        abbreviated as m).
+    random_seed:  A common random seed that is used for
+        creating all Liquid Legions sketches.  This determines the hash
+        function.  Any two sketches that might need to be combined need
+        to be created with the same seed.
+    """
+
+    decay_rate: int = 12
+    sketch_size: int = int(1e5)
+    random_seed: int = 1
+
+
+class SimulationParameters(NamedTuple):
+    """Parameters defining a simulation run.
+
+    Attributes:
+        campaign_spend:  A list of spend values, one per campaign,
+            representing the total actual spend for each campaign.
```

Comment:
Why a list of total spends instead of a list of spend_vector? Are you suggesting random split of the total spend per campaign.

Of course, it depends on how we're going to `self._params` in `HaloSimulator` and `PlannerSimulator`.

Reply:
A PublisherData object represents the total potential inventory that a publisher can serve.  This would include impressions that were actually served during the campaign plus impressions that could have potentially been served if the spend had been higher.  The campaign_spend specifies that actual amount that was spent during the campaign.  From the standpoint of the Halo Simulator, it is not able to return simulated reach values if the spend is above the campaign_spend.  These must be extrapolated.  One thing that we would like to test is how well the different models perform at extrapolation.  I hope this makes sense.

Reply:
Got it! A follow-up question to check if I understand correctly.

Now we define a testing point as an extrapolation (against campaign A) if sum(spend_vector(testing_point)) > sum(spend_vector(campaign A)). 

An alternative condition is "not spend_vector(testing_point) <= spend_vector(campaign A) ", i.e., as long as the spend of the testing point exceeds the actual spend of campaign A at **one publisher**, the point is viewed as an extrapolation. However, this alternative definition is not viable since we do not know the spend vector of each campaign --- we only know the total spend. Is that correct? If there's confusion let's chat offline.

Reply:
You are right, if the spend with any publisher is higher than the campaign_spend for that publisher, then the point should be viewed as an extrapolation.  Each campaign runs on a single publisher.  So the spend for that campaign is represented by a scalar value.  The spend across all campaigns (e.g., the campaign_spend field above) can be given by a vector of values equal in length to the number of publishers.